### ISSUE TITLE
error getting device state SHC1

### ISSUE THREAD

**Issue by metaiiica** [planbnet/livisi_unofficial#23]
### System Health details

Device is a empty Testsystem (Linux Mint with HA 8.4 in VM installed), only Hacs, SSH, Samba and Fileeditor installed. Integration only Livisi unofficial. 

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [previous issues](https://github.com/ludeeus/integration_blueprint/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

see Screenshot. Many Value are not loaded

https://ibb.co/xj16zXN

### Reproduction steps

1. install Livisi unofficial 1.31
2. reboot HA

### Debug logs

```text
Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: custom_components.livisi
Source: custom_components/livisi/aiolivisi.py:122
Integration: Livisi Unofficial (documentation, issues)
First occurred: 01:04:16 (130 occurrences)
Last logged: 01:12:34

Error getting device state
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 980, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1085, in create_connection
    raise exceptions[0]
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1069, in create_connection
    sock = await self._connect_sock(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 973, in _connect_sock
    await self.sock_connect(sock, address)
  File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 628, in sock_connect
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 668, in _sock_connect_cb
    raise OSError(err, f'Connect call failed {address}')
ConnectionRefusedError: [Errno 111] Connect call failed ('192.168.178.124', 8080)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/livisi/aiolivisi.py", line 222, in async_get_device_state
    return await self.async_send_authorized_request(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/livisi/aiolivisi.py", line 84, in async_send_authorized_request
    return await self.async_send_request(method, url, payload, self._auth_headers)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/livisi/aiolivisi.py", line 122, in async_send_request
    response = await self._async_send_request(method, url, payload, headers)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 536, in _request
    conn = await self._connector.connect(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 540, in connect
    proto = await self._create_connection(req, traces, timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 901, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 1209, in _create_direct_connection
    raise last_exc
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 1178, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 988, in _wrap_create_connection
    raise client_error(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host 192.168.178.124:8080 ssl:False [Connect call failed ('192.168.178.124', 8080)]
```

**Comment by planbnet** [comment]
@DanielHabenicht Your fix seems to also fail on the second try here.., Does it work for you?

The only reason I can think of is that the SHC v1 can only process one request in parallel, so this commit might be responsible
https://github.com/planbnet/livisi_unofficial/commit/d012b3a3ceeddf4cde4589880d495e348991b554

**Comment by DanielHabenicht** [comment]
I have no errors in my log. 
But many devices show unknown values for the first few minutes after the restart of the SHC1. 

For the Humidity and Temperature readings I also often get unknown values. I haven't looked into why yet, but it seemingly has to do with the double entities for the room and the thermostat itself.

**Comment by planbnet** [comment]
This really sounds like a problem with parallel requests. On my v2 SHC everything works correctly. When initializing the devices, many requests at once will be sent (because the devices are initialized asynchonously by hass). Probably in a v1 SHC with many devices some of those requests fail and the values only appear as soon as an update is sent via the websocket.
As my integration creates so much more entities, the problem is much more likely to appear here but theoretically also possible on the official integration...

**Comment by planbnet** [comment]
But now it happens when setting the thermostat temperature???? Is the rest working now?

**Comment by DanielHabenicht** [comment]
@planbnet I tested parallel requests on the SHC 1 and this does not seem to be an issue. I was able to request 14 URLs in parallel with no issue: 

![image](https://github.com/planbnet/livisi_unofficial/assets/13590797/0780e210-6a7b-4590-9c38-dec68fb0a95d)

More are possible but it gets flaky then:

![image](https://github.com/planbnet/livisi_unofficial/assets/13590797/710998bb-d979-449c-84fa-a975aa4962bb)

Same picture for querying all capabilities at the same time: 
![image](https://github.com/planbnet/livisi_unofficial/assets/13590797/4ea93010-a302-4857-8d0d-2e4ba0c95114)

**Comment by DanielHabenicht** [comment]
In Home Assistant I can't reproduce the specific error mentioned above while setting my thermostat (but still `Server disconnected` errors).  In addition, I get this one: 
```
Traceback (most recent call last):
  File "/config/custom_components/livisi/aiolivisi.py", line 222, in async_get_device_state
    return await self.async_send_authorized_request(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/livisi/aiolivisi.py", line 84, in async_send_authorized_request
    return await self.async_send_request(method, url, payload, self._auth_headers)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/livisi/aiolivisi.py", line 124, in async_send_request
    if "errorcode" in response:
       ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'NoneType' is not iterable
```

**Comment by metaiiica** [comment]
here the Screenshot

[Screenshot](https://ibb.co/4tB9Sf6)

**Comment by DanielHabenicht** [comment]
Are the thermostates of different types?

**Comment by planbnet** [comment]
@DanielHabenicht I don't see how you come to the conclusion that parallel requests are _not_ the problem when you can even reliably reproduce them above 14. When you have a few devices, initializing them in parallel can easily create more requests. Probably just my rate limiting code does not work.

Please try the following in your test code and play with the values of limit and limit_per_host (which i currently do not set):
```
async def main(urls):
    async with aiohttp.ClientSession(connector=TCPConnector(
            limit=1, limit_per_host=1
        )) as session:
```

**Comment by metaiiica** [comment]
> Are the thermostates of different types?

no, all the same

**Comment by metaiiica** [comment]
in the Livisiforum i have asked if any give you a SHC1 if needet to reproduce the issue what i have... so it have better  chance to fix all issue... 

https://community.livisi.de/forum/thread/4900-warum-verliert-ha-den-connect-zum-shc-nach-einiger-zeit/?postID=42188#post42190

**Comment by metaiiica** [comment]
Please let us write in this forum, here are more user and we can write in german so it is better for all

**Comment by planbnet** [comment]
Sorry, I have no interest to use that forum. Github is fine for me and the project description states: "This project is in "works for me" state and I do not advise anyone to use it nor can or will I provide support on how to install it." 
I don't have the time and motivation to act as an unpaid Livisi supporter for frustrated users that will soon lose access to their devices.

I already own an old SHC v1, and if I find the time and feel bored I might setup it to debug this issue, but at the moment this is not a very high priority for me.

Besides that I'll stick to english here, because the home assistant team should be able to understand the issues and PRs in this repo in case the code will someday be merged back to the official integration.

**Comment by planbnet** [comment]
Closing this because after the 1.4.0 rewrite and the 1.4.1 fix for SHC v1 this still might be a problem, but would materialize differently. If you still experience problems with the thermostats, please open a new issue.
